cap/liquidity.py

```diff
@@ -0,0 +1,55 @@
+from utils.abi import load_abi
+from utils.chains import Chain
+from utils.telegram import send_telegram_message
+from utils.web3_wrapper import ChainManager
+
+CUSD = "0xcCcc62962d17b8914c62D74FfB843d73B2a3cccC"
+PROTOCOL = "cap"
+
+
+def main():
+    client = ChainManager.get_client(Chain.MAINNET)
+
+    ctoken = client.eth.contract(address=CUSD, abi=load_abi("cap/abi/CToken.json"))  # aka cusd
+
+    assets = ctoken.functions.assets().call()
+
+    # Batch 1: resolve vault addresses for each asset
+    with client.batch_requests() as batch:
+        for asset in assets:
+            batch.add(ctoken.functions.fractionalReserveVault(asset))
+        vault_addresses = batch.execute()
+
+    # Batch 2: for each asset, get vault maxWithdraw for CUSD owner, token balance, decimals, and symbol
+    with client.batch_requests() as batch:
+        for asset, vault_addr in zip(assets, vault_addresses):
+            vault = client.eth.contract(address=vault_addr, abi=load_abi("cap/abi/YearnV3Vault.json"))
+            token = client.eth.contract(address=asset, abi=load_abi("common-abi/ERC20.json"))
```

Comment:
Why do you need to fetch `token` here? Token should be the same as asset?! `token==asset`?

Reply:
asset.address == token.address

returnt "asset" is not an eth_contract instance but "token" is